pal/inc/safemath.h

```diff
@@ -156,717 +167,6 @@ inline bool DoubleFitsInIntType(double val)
 
 #define ovadd3_gt(a, b, c, rhs) (((a) + (b) + (c) > (rhs)) || ((a) + (b) < (a)) || ((a) + (b) + (c) < (c)))
 
-
- //-----------------------------------------------------------------------------
-//
-// Liberally lifted from the Office example on MSDN and modified.
-// http://msdn.microsoft.com/library/en-us/dncode/html/secure01142004.asp
-//
-// Modified to track an overflow bit instead of throwing exceptions.  In most
-// cases the Visual C++ optimizer (Whidbey beta1 - v14.00.40607) is able to 
-// optimize the bool away completely.
-// Note that using a sentinal value (IntMax for example) to represent overflow
-// actually results in poorer code-gen.
-//
-// This has also been simplified significantly to remove functionality we 
-// don't currently want (division, implicit conversions, many additional operators etc.)
-//
-// Example:
-//   unsafe: UINT32 bufSize = headerSize + elementCount * sizeof(void*);
-//   becomes:
-//      S_UINT32 bufSize = S_UINT32(headerSize) + S_UINT32(elementCount) *
-//                                          S_UINT32( sizeof(void*) );
-//      if( bufSize.IsOverflow() ) { <overflow-error> }
-//      else { use bufSize.Value() }
-//   or:
-//      UINT32 tmp, bufSize;
-//      if( !ClrSafeInt<UINT32>::multiply( elementCount, sizeof(void*), tmp ) ||
-//              !ClrSafeInt<UINT32>::addition( tmp, headerSize, bufSize ) ) 
-//      { <overflow-error> }
-//      else { use bufSize }
-//      
-//-----------------------------------------------------------------------------
-// TODO: Any way to prevent unintended instantiations?  This is only designed to
-//  work with unsigned integral types (signed types will work but we probably 
-//  don't need signed support).
-template<typename T> class ClrSafeInt
```

Comment:
Was it necessary to remove ClrSafeInt because of SDK versions or was this a drive-by cleanup?